src/tree/ResolvedContainerAppsResource.ts

```diff
@@ -0,0 +1,176 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ContainerApp, ContainerAppsAPIClient, ManagedEnvironment } from "@azure/arm-appcontainers";
+import { LocationListStep, uiUtils, VerifyProvidersStep } from "@microsoft/vscode-azext-azureutils";
+import { AzExtTreeItem, AzureWizard, AzureWizardExecuteStep, AzureWizardPromptStep, DialogResponses, IActionContext, ICreateChildImplContext, ISubscriptionContext, parseError, UserCancelledError } from "@microsoft/vscode-azext-utils";
+import { ResolvedAppResourceBase } from "@microsoft/vscode-azext-utils/hostapi";
+import { ProgressLocation, window } from "vscode";
+import { ContainerAppCreateStep } from "../commands/createContainerApp/ContainerAppCreateStep";
+import { ContainerAppNameStep } from "../commands/createContainerApp/ContainerAppNameStep";
+import { EnableIngressStep } from "../commands/createContainerApp/EnableIngressStep";
+import { EnvironmentVariablesListStep } from "../commands/createContainerApp/EnvironmentVariablesListStep";
+import { IContainerAppWithActivityContext } from "../commands/createContainerApp/IContainerAppContext";
+import { ContainerRegistryListStep } from "../commands/deployImage/ContainerRegistryListStep";
+import { azResourceContextValue, webProvider } from "../constants";
+import { ext } from "../extensionVariables";
+import { createActivityContext } from "../utils/activityUtils";
+import { createContainerAppsAPIClient } from "../utils/azureClients";
+import { getResourceGroupFromId } from "../utils/azureUtils";
+import { localize } from "../utils/localize";
+import { nonNullProp } from "../utils/nonNull";
+import { settingUtils } from "../utils/settingUtils";
+import { ContainerAppTreeItem } from "./ContainerAppTreeItem";
+import { IAzureResource } from "./IAzureResourceTreeItem";
+import { ManagedEnvironmentTreeItem } from "./ManagedEnvironmentTreeItem";
+
+export class ResolvedContainerAppsResource implements ResolvedAppResourceBase, IAzureResource {
```

Comment:
It's a little confusing that the class is a `ResolvedContainerAppsResource` but is actually a `containerEnvironment`. I think calling it a `ResolvedContainerEnvironmentResource` might make more sense in this case even if the ext is Container Apps.